cosmos/operators/local.py

```diff
@@ -252,6 +256,67 @@ def store_compiled_sql(self, tmp_project_dir: str, context: Context, session: Se
         else:
             self.log.info("Warning: ti is of type TaskInstancePydantic. Cannot update template_fields.")
 
+    @staticmethod
+    def _configure_remote_target_path() -> tuple[Path, str] | tuple[None, None]:
+        """Configure the remote target path if it is provided."""
+        if not remote_target_path:
+            return None, None
+
+        _configured_target_path = None
+
+        target_path_str = str(remote_target_path)
+
+        remote_conn_id = remote_target_path_conn_id
+        if not remote_conn_id:
+            target_path_schema = target_path_str.split("://")[0]
+            remote_conn_id = FILE_SCHEME_AIRFLOW_DEFAULT_CONN_ID_MAP.get(target_path_schema, None)  # type: ignore[assignment]
+        if remote_conn_id is None:
+            return None, None
+
+        if not AIRFLOW_IO_AVAILABLE:
+            raise CosmosValueError(
+                f"You're trying to specify remote target path {target_path_str}, but the required "
+                f"Object Storage feature is unavailable in Airflow version {airflow_version}. Please upgrade to "
+                "Airflow 2.8 or later."
+            )
+
+        from airflow.io.path import ObjectStoragePath
+
+        _configured_target_path = ObjectStoragePath(target_path_str, conn_id=remote_conn_id)
+
+        if not _configured_target_path.exists():  # type: ignore[no-untyped-call]
+            _configured_target_path.mkdir(parents=True, exist_ok=True)
+
+        return _configured_target_path, remote_conn_id
+
+    def upload_compiled_sql(self, tmp_project_dir: str, context: Context) -> None:
+        """
+        Uploads the compiled SQL files from the dbt compile output to the remote store.
+        """
+        if not self.should_upload_compiled_sql:
+            return
+
+        dest_target_dir, dest_conn_id = self._configure_remote_target_path()
+        if not dest_target_dir:
+            raise CosmosValueError(
+                "You're trying to upload compiled SQL files, but the remote target path is not configured. "
+            )
+
+        from airflow.io.path import ObjectStoragePath
+
+        source_compiled_dir = Path(tmp_project_dir) / "target" / "compiled"
+        files = [str(file) for file in source_compiled_dir.rglob("*") if file.is_file()]
+
+        for file_path in files:
+            rel_path = os.path.relpath(file_path, source_compiled_dir)
+
+            dest_path = ObjectStoragePath(
+                f"{str(dest_target_dir).rstrip('/')}/{context['dag'].dag_id}/{rel_path.lstrip('/')}",
```

Comment:
Let's isolate the logic to build this URL in a function. It should also take into account the `TaskGroup` ID, since the same DAG may potentially have multiple TaskGroups referencing different dbt project paths.